server/db/stats_store.go

```diff
@@ -0,0 +1,329 @@
+	s.metrics.IncStoreOp("GetAvgCallParticipants")
+	defer func(start time.Time) {
+		s.metrics.ObserveStoreMethodsTime("GetAvgCallParticipants", time.Since(start).Seconds())
+	}(time.Now())
+
+	qb := getQueryBuilder(s.driverName).
+		Select("AVG(jsonb_array_length(participants))").
+		From("calls").
+		Where("jsonb_typeof(participants) = 'array'")
+	if s.driverName == model.DatabaseDriverMysql {
+		qb = getQueryBuilder(s.driverName).Select("AVG(json_length(participants))").From("calls")
+	}
+	qb = qb.Where(sq.And{
+		sq.Expr("EndAt > StartAt"),
+		sq.Eq{"DeleteAt": 0},
+	})
+
+	q, args, err := qb.ToSql()
+	if err != nil {
+		return 0, fmt.Errorf("failed to prepare query: %w", err)
+	}
+
+	var count *float64
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(*s.settings.QueryTimeout)*time.Second)
+	defer cancel()
+	if err := s.rDBx.GetContext(ctx, &count, q, args...); err != nil {
+		return 0, fmt.Errorf("failed to get average call participants: %w", err)
+	}
+
+	if count == nil {
+		return 0, nil
+	}
+
+	return int64(math.Round(*count)), nil
+}
+
+func (s *Store) GetAvgCallDuration() (int64, error) {
+	s.metrics.IncStoreOp("GetAvgCallDuration")
+	defer func(start time.Time) {
+		s.metrics.ObserveStoreMethodsTime("GetAvgCallDuration", time.Since(start).Seconds())
+	}(time.Now())
+
+	qb := getQueryBuilder(s.driverName).Select("AVG(EndAt - StartAt)/1000").
+		From("calls").
+		Where(sq.And{
+			sq.Expr("EndAt > StartAt"),
+			sq.Eq{"DeleteAt": 0},
+		})
+
+	q, args, err := qb.ToSql()
+	if err != nil {
+		return 0, fmt.Errorf("failed to prepare query: %w", err)
+	}
+
+	var count *float64
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(*s.settings.QueryTimeout)*time.Second)
+	defer cancel()
+	if err := s.rDBx.GetContext(ctx, &count, q, args...); err != nil {
+		return 0, fmt.Errorf("failed to get average call duration: %w", err)
+	}
+
+	if count == nil {
+		return 0, nil
+	}
+
+	return int64(math.Round(*count)), nil
+}
+
+func (s *Store) GetTotalActiveSessions() (int64, error) {
+	s.metrics.IncStoreOp("GetTotalActiveSessions")
+	defer func(start time.Time) {
+		s.metrics.ObserveStoreMethodsTime("GetTotalActiveSessions", time.Since(start).Seconds())
+	}(time.Now())
+
+	qb := getQueryBuilder(s.driverName).Select("COUNT(*)").
+		From("calls_sessions").
+		Join("calls ON calls_sessions.CallID = calls.ID").
+		Where(sq.And{
+			sq.Eq{"calls.EndAt": 0},
+			sq.Eq{"calls.DeleteAt": 0},
+		})
+
+	q, args, err := qb.ToSql()
+	if err != nil {
+		return 0, fmt.Errorf("failed to prepare query: %w", err)
+	}
+
+	var count int64
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(*s.settings.QueryTimeout)*time.Second)
+	defer cancel()
+	if err := s.rDBx.GetContext(ctx, &count, q, args...); err != nil {
+		return 0, fmt.Errorf("failed to get calls sessions count: %w", err)
+	}
+
+	return count, nil
+}
+
+func (s *Store) GetTotalCalls(active bool) (int64, error) {
```

Comment:
Keeping it simple for now, but ideally, most of these methods could filter for the last month or so. That way they would perform much better.